### PR TITLE
SDKS-3463_Create OIDC login sample app

### PR DESCRIPTION
**Pull request by george-bafaloukas-forgerock**
SDKS-3463_Create OIDC login sample app
Screenshot 2024-11-26 at 16 28 09
Screenshot 2024-11-26 at 16 28 05

### CODE REVIEW

File: iOS/swiftui-oidc/PingExample/PingExample/ViewModels/ConfigurationViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+class ConfigurationViewModel: ObservableObject {
+    
+    @Published public var clientId: String
+    @Published public var scopes: [String]
+    @Published public var redirectUri: String
+    @Published public var signOutUri: String?
+    @Published public var discoveryEndpoint: String
+    @Published public var environment: String
+    @Published public var cookieName: String?
+    @Published public var browserSeletorType: BrowserSeletorTypes
```
Comment (jeyanthanperiyasamy) [inline iOS/swiftui-oidc/PingExample/PingExample/ViewModels/ConfigurationViewModel.swift:23]: browserSelectorType - check spelling

File: iOS/swiftui-oidc/PingExample/PingExample/UserProfileViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    @Published var userProfile: String = ""
+    
+    init() {
+        Task {
+            await self.accessToken()
+        }
+    }
+    
+    func accessToken() async {
+//        let foo = await davinci.user()?.userinfo(cache: <#T##Bool#>)
```
Comment (rodrigoareis) [inline iOS/swiftui-oidc/PingExample/PingExample/UserProfileViewModel.swift:21]: Can we provide details here why this is commented out?

File: iOS/swiftui-oidc/PingExample/PingExample/Utilities/ConfigurationManager.swift [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        let defaults = UserDefaults.standard
+        if let savedConfiguration = defaults.object(forKey: "CurrentConfiguration") as? Data {
+            let decoder = JSONDecoder()
+            if let loadedConfiguration = try? decoder.decode(Configuration.self, from: savedConfiguration) {
+                return ConfigurationViewModel(clientId: loadedConfiguration.clientId, scopes: loadedConfiguration.scopes, redirectUri: loadedConfiguration.redirectUri, signOutUri: loadedConfiguration.signOutUri, discoveryEndpoint: loadedConfiguration.discoveryEndpoint, environment: loadedConfiguration.environment, cookieName: loadedConfiguration.cookieName, browserSeletorType: BrowserSeletorTypes(rawValue: loadedConfiguration.browserType.description) ?? BrowserSeletorTypes.authSession)
+            }
+        }
+        
+        return ConfigurationViewModel(
+            clientId: "ForgeRockSDKClient",
```
Comment (rodrigoareis) [inline iOS/swiftui-oidc/PingExample/PingExample/Utilities/ConfigurationManager.swift:93]: I suggest to replace this valid config with a placeholder config

File: iOS/swiftui-oidc/PingExample/PingExample/Utilities/ConfigurationManager.swift [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            signOutUri: "org.forgerock.demo://oauth2redirect",
+            discoveryEndpoint: "https://openam-bafaloukas.forgeblocks.com/am/oauth2/realms/alpha/.well-known/openid-configuration",
+            environment: "AIC",
+            cookieName: "386c0d288cac4b9",
+            browserSeletorType: .authSession
+        )
+    }
+}
+
+/*
```
Comment (rodrigoareis) [inline iOS/swiftui-oidc/PingExample/PingExample/Utilities/ConfigurationManager.swift:105]: Remove this?

File: iOS/swiftui-oidc/PingExample/PingExample/Utilities/ConfigurationManager.swift [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    public func defaultConfigurationViewModel() -> ConfigurationViewModel {
+        let defaults = UserDefaults.standard
+        if let savedConfiguration = defaults.object(forKey: "CurrentConfiguration") as? Data {
+            let decoder = JSONDecoder()
+            if let loadedConfiguration = try? decoder.decode(Configuration.self, from: savedConfiguration) {
+                return ConfigurationViewModel(clientId: loadedConfiguration.clientId, scopes: loadedConfiguration.scopes, redirectUri: loadedConfiguration.redirectUri, signOutUri: loadedConfiguration.signOutUri, discoveryEndpoint: loadedConfiguration.discoveryEndpoint, environment: loadedConfiguration.environment, cookieName: loadedConfiguration.cookieName, browserSeletorType: BrowserSeletorTypes(rawValue: loadedConfiguration.browserType.description) ?? BrowserSeletorTypes.authSession)
+            }
+        }
+        
+        return ConfigurationViewModel(
```
Comment (rodrigoareis) [inline iOS/swiftui-oidc/PingExample/PingExample/Utilities/ConfigurationManager.swift:92]: I would suggest to add a TODO here. Something like:

`//TODO: Provide here the client configuration. Replace the placeholder with the OAuth2.0 client details`

**Review by rodrigoareis — CHANGES_REQUESTED** [review]
Left some comments to address. Please add a README file to the app folder. You can follow the structure of the `koltin-davinci` app.

File: iOS/swiftui-oidc/PingExample/PingExample/Utilities/ConfigurationManager.swift (hunk 3 above)
Comment (george-bafaloukas-forgerock) [reply iOS/swiftui-oidc/PingExample/PingExample/Utilities/ConfigurationManager.swift:93]: yeah those are leftovers, will remove and add a placeholder

File: iOS/swiftui-oidc/PingExample/PingExample/UserProfileViewModel.swift (hunk 2 above)
Comment (george-bafaloukas-forgerock) [reply iOS/swiftui-oidc/PingExample/PingExample/UserProfileViewModel.swift:21]: No idea...left over I guess...I will remove

**Review by rodrigoareis — APPROVED** [review]
Changes looks great.

**Review by vahancouver — CHANGES_REQUESTED** [review]
Here are my comments
    1. LogoView, UsernameTextField, PasswordTextField, LoginButton, HavingTrouble, RegisterNow, ActivityIndicatorView structs defined in LoginUitility class are not used anywhere in the app. Must be removed from codebase
    2. View Model object naming is not consistent throughout the project (we have cm, viewmodel, xxxYYYViewModel namings). Ideally I would name every view model object same as the class (starting with lowercase). For Example userInfoViewModel.
    3. We have TokenViewModel and AccessTokenView. Ideally I would name the first one AccessTokenViewModel for consistency 
    4. ErrorView and HeaderView structs defined in LoginView file are not used anywhere. Should be removed. However any reason we are not using the ErrorView to show the errors?
    5. There are extra empty lines here and there. I would go thru all the files and make sure the formatting is consistent everywhere.
    6. The button color needs to be changed to Ping Red: Color(red: 163.0/255.0, green: 19.0/255.0, blue: 0.0/255.0)

**Comment by george-bafaloukas-forgerock** [comment]
> ```
> Here are my comments
> 1. LogoView, UsernameTextField, PasswordTextField, LoginButton, HavingTrouble, RegisterNow, ActivityIndicatorView structs defined in LoginUitility class are not used anywhere in the app. Must be removed from codebase
> 2. View Model object naming is not consistent throughout the project (we have cm, viewmodel, xxxYYYViewModel namings). Ideally I would name every view model object same as the class (starting with lowercase). For Example userInfoViewModel.
> 3. We have TokenViewModel and AccessTokenView. Ideally I would name the first one AccessTokenViewModel for consistency 
> 4. ErrorView and HeaderView structs defined in LoginView file are not used anywhere. Should be removed. However any reason we are not using the ErrorView to show the errors?
> 5. There are extra empty lines here and there. I would go thru all the files and make sure the formatting is consistent everywhere.
> 6. The button color needs to be changed to Ping Red: Color(red: 163.0/255.0, green: 19.0/255.0, blue: 0.0/255.0)
> ```

Hey good points. Will have a look and do a clean up. The error view and some of those were relevant mostly on the Davinci sample.

File: iOS/swiftui-oidc/PingExample/PingExample/Views/ContentView.swift
Comment (spetrov) [inline iOS/swiftui-oidc/PingExample/PingExample/Views/ContentView.swift:1]: Missing copyright header...

**Review by spetrov — APPROVED** [review]
Looks good to me! 👍🏻

**Pull request merged**